lightning/src/ln/channel.rs

```diff
@@ -3058,6 +3098,31 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let keys = if let Ok(keys) = self.build_holder_transaction_keys(self.cur_holder_commitment_transaction_number) { keys } else { return None; };
+		let commitment_stats = self.build_commitment_transaction(self.cur_holder_commitment_transaction_number, &keys, true, true, logger);
+		let buffer_fee_msat = Channel::<Signer>::commit_tx_fee_sat(feerate_per_kw, commitment_stats.num_nondust_htlcs + outbound_stats.holding_cell_htlcs_count as usize + CONCURRENT_INBOUND_HTLC_FEE_BUFFER as usize) * 1000;
```

Comment:
Don't we want to filter `holding_cell_htlcs_count` to only include nondust HTLCs?

Reply:
Right, though I think it's the good behavior to account the dust HTLC in `holding_cell_msat` as they're effectively retrieved from the sender balance, and not available anymore to feed the fee burden ?

Reply:
True. In general I don't think we need to be *super* careful about accounting for holding cell HTLCs here as we should be doing the tests again before we actually commit to the HTLC, so if we fail to send it then its not ideal, but won't cause a force-close.